package/forms.py

```diff
@@ -195,14 +197,70 @@ def save(self, *args, **kwargs):
 )
 
 
-class ProjectImagesFormSet(BaseProjectImagesFormSet):
+class ProjectImageUrlForm(forms.Form):
 
-    def __init__(self, project, queryset=None, *args, **kwargs):
-        self.project = project
+    url = forms.URLField()
+    project = forms.CharField()
+
+    def __init__(self, project, *args, **kwargs):
+        super(ProjectImageUrlForm, self).__init__(*args, **kwargs)
+        self.image_path = None
+        self.absolute_image_path = None
+        self.delete = False
+        self.fields["url"].widget = forms.HiddenInput()
+        self.fields["project"].widget = forms.HiddenInput()
+        self.fields["project"].initial = project
+
+    def clean(self):
+        cleaned_data = super(ProjectImageUrlForm, self).clean()
+        image_url = cleaned_data.get("url")
+        project = cleaned_data.get("project")
+        self.delete = cleaned_data.get("DELETE")
+        if self.delete:
+            try:
+                self.absolute_image_path = cut_domain_name_from_url(image_url)
+                delete_file_from_media(self.absolute_image_path)
+                self.image_path = self.absolute_image_path.split("/", 1)[-1]
+            except ValidationError:
```

Comment:
which of those 3 lines in `try` block can throw this exception? If only one line among them, then only one line should be placed in `try` segment